x/twap/migrate_test.go

```diff
@@ -0,0 +1,45 @@
+package twap_test
+
+import (
+	"time"
+
+	"github.com/osmosis-labs/osmosis/v11/x/twap/types"
+)
+
+func (s *TestSuite) TestMigrateExistingPools() {
+	// create two pools before migration
+	s.PrepareUni2PoolWithAssets(defaultUniV2Coins[0], defaultUniV2Coins[1])
+	s.PrepareBalancerPool()
+
+	// suppose upgrade happened and increment block height and block time
+	s.Ctx = s.Ctx.WithBlockHeight(s.Ctx.BlockHeight() + 1)
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(time.Second * 10))
+
+	// run migration logic
+	latestPoolId := s.App.GAMMKeeper.GetNextPoolId(s.Ctx) - 1
```

Comment:
I think it would be valuable to also have a test with `s.App.GAMMKeeper.GetNextPoolId(s.Ctx)` (no -1) and expect that to error

Reply:
done!